### PR TITLE
Use GetCurrentManifestPath to locate current MANIFEST file

### PR DESCRIPTION
**Pull request by HaoyuHuang**
In version_set.cc, there is a function GetCurrentManifestPath. The goal of this task is to refactor ListColumnFamilies function so that ListColumnFamilies calls GetCurrentManifestPath to search for MANIFEST.

Test plan
$make clean && make -j20 all && sleep 1 && make check
All tests must pass without failure.

### CODE REVIEW

File: db/version_set.cc [hunk 1]
```diff
@@ -5510,7 +5508,8 @@ Status ReactiveVersionSet::MaybeSwitchManifest(
   Status s;
   do {
     std::string manifest_path;
-    s = GetCurrentManifestPath(&manifest_path);
+    s = GetCurrentManifestPath(&manifest_path, &manifest_file_number_,
```
Comment (riversand963) [inline db/version_set.cc]: Can you run `make format` to adjust the format?

File: db/version_set.cc [hunk 2]
```diff
@@ -4041,10 +4041,11 @@ Status VersionSet::ExtractInfoFromVersionEdit(
   return Status::OK();
 }
 
-Status VersionSet::GetCurrentManifestPath(std::string* manifest_path) {
+Status VersionSet::GetCurrentManifestPath(std::string* manifest_path,
+  uint64_t* manifest_file_number, const std::string& dbname, Env* env) {
```
Comment (riversand963) [inline db/version_set.cc]: Nit: assert `manifest_file_number != nullptr` as well. Same for `env`.

File: db/version_set.cc [hunk 3]
```diff
@@ -4041,10 +4041,11 @@ Status VersionSet::ExtractInfoFromVersionEdit(
   return Status::OK();
 }
 
-Status VersionSet::GetCurrentManifestPath(std::string* manifest_path) {
+Status VersionSet::GetCurrentManifestPath(std::string* manifest_path,
```
[the remark this answers is not in the document]
Comment (HaoyuHuang) [reply db/version_set.cc]: I updated the order.

File: db/version_set.cc (hunk 2 above)
Comment (HaoyuHuang) [reply db/version_set.cc]: Added the asserts.

File: db/version_set.cc (hunk 1 above)
Comment (HaoyuHuang) [reply db/version_set.cc]: Sure.

**Pull request closed without merging**